components/eamxx/src/share/field/field_impl_details.hpp

```diff
@@ -0,0 +1,269 @@
+#ifndef SCREAM_FIELD_IMPL_DETAILS_HPP
+#define SCREAM_FIELD_IMPL_DETAILS_HPP
+
+#include <ekat/kokkos/ekat_kokkos_types.hpp>
+
+#include <vector>
+
+namespace scream
+{
+
+namespace details {
+
+template<CombineMode CM, bool use_fill, typename LhsView, typename RhsView, typename ST>
+struct CombineViewsHelper {
+
+  using exec_space = typename LhsView::traits::execution_space;
+
+  static constexpr int N = LhsView::rank();
+
+  template<int M>
+  using MDRange = Kokkos::MDRangePolicy<
+                    exec_space,
+                    Kokkos::Rank<M,Kokkos::Iterate::Right,Kokkos::Iterate::Right>
+                  >;
+
+  void run (const std::vector<int>& dims) const {
+    if constexpr (N==0) {
+      Kokkos::RangePolicy<exec_space> policy(0,1);
+      Kokkos::parallel_for(policy,*this);
+    } else if constexpr (N==1) {
+      Kokkos::RangePolicy<exec_space> policy(0,dims[0]);
+      Kokkos::parallel_for(policy,*this);
+    } else if constexpr (N==2) {
+      MDRange<2> policy({0,0},{dims[0],dims[1]});
```

Comment:
I think @jgfouca found that `RangePolicy` with a flattened multi-index was significantly faster than `MDRange` inside rrtmgp (is that correct?). I wonder if we should test that here?

Reply:
I did ask kokkos folks about this, and they said MD range ought to be as performant as Range with md index flattening, at least for large enough dimensions. [Here](https://kokkosteam.slack.com/archives/C5BGU5NDQ/p1738692643588899) is the convo on the kokkos slack space.

I don't know why in rrtmgp MDRange was slower. I wonder if it had to do with the views dimension. But here, I would really really like to keep MDRange, as it is way more expressive than Range, and also much less bug prone (less chance of index arithmetic mistakes).